File: tks/ApplicationCore/DomainModel/src/test/kotlin/model/values/AccessionNumberTest.java
package model.values;

import exceptions.TypeValidationFailedException;
import lombok.SneakyThrows;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class AccessionNumberTest {

    @Test
    public void validationTest(){
        assertThrows(TypeValidationFailedException.class, () -> new AccessionNumber("xxx"));
        assertThrows(TypeValidationFailedException.class, () -> new AccessionNumber("AAAA-5555"));
        assertThrows(TypeValidationFailedException.class, () -> new AccessionNumber("AAAA-55"));
        assertDoesNotThrow(() -> new AccessionNumber("SCXI-591"));
    }

    @SneakyThrows
    @Test
    public void equalityTest(){
        var accessionNumber1 = new AccessionNumber("SCXI-591");
        var accessionNumber2 = new AccessionNumber("SCXI-591");
        var accessionNumber3 = new AccessionNumber("SCXI-592");

        assertNotSame(accessionNumber1, accessionNumber2);
        assertEquals(accessionNumber1, accessionNumber2);
        assertNotEquals(accessionNumber1, accessionNumber3);
    }

}

File: tks/core/domain-model/domain-common/src/test/kotlin/core/domain/common/valueobjects/AccessionNumberTest.kt
@file:Suppress("SpellCheckingInspection")

package core.domain.common.valueobjects

import core.domain.common.exceptions.TypeValidationFailedException
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class AccessionNumberTest {
    @Test
    fun validationTest() {
        Assertions.assertThrows(core.domain.common.exceptions.TypeValidationFailedException::class.java) {
            core.domain.common.valueobjects.AccessionNumber(
                "xxx"
            )
        }
        Assertions.assertThrows(core.domain.common.exceptions.TypeValidationFailedException::class.java) {
            core.domain.common.valueobjects.AccessionNumber(
                "AAAA-5555"
            )
        }
        Assertions.assertThrows(core.domain.common.exceptions.TypeValidationFailedException::class.java) {
            core.domain.common.valueobjects.AccessionNumber(
                "AAAA-55"
            )
        }
        Assertions.assertDoesNotThrow<core.domain.common.valueobjects.AccessionNumber> {
            core.domain.common.valueobjects.AccessionNumber(
                "SCXI-591"
            )
        }
    }

    @Test
    fun equalityTest() {
        val accessionNumber1 = core.domain.common.valueobjects.AccessionNumber("SCXI-591")
        val accessionNumber2 = core.domain.common.valueobjects.AccessionNumber("SCXI-591")
        val accessionNumber3 = core.domain.common.valueobjects.AccessionNumber("SCXI-592")
        Assertions.assertNotSame(accessionNumber1, accessionNumber2)
        Assertions.assertEquals(accessionNumber1, accessionNumber2)
        Assertions.assertNotEquals(accessionNumber1, accessionNumber3)
    }
}